src/core/funcbloc.h

```diff
@@ -604,6 +623,15 @@ class CFunctionBlock : public forte::core::CFBContainer {
 
     const SFBInterfaceSpec &mInterfaceSpec; //!< Pointer to the interface specification
 
+    /*!\brief initialize the data structure which holds connection counts per pin
+    * \param paNumEIs number of eventInputs
+    */
+    void setupInputConnectionTrackingData(TEventID paNumEIs) {
+      if (getFBInterfaceSpec().mEITypeNames != nullptr) {
+        mInputEventConnectionCount = std::make_unique<size_t[]>(paNumEIs);
+      }
+    }
```

Comment:
Please also move the `getFBInterfaceSpec().mNumEIs` argument into `setupInputConnectionTrackingData()`:
```suggestion
    void setupInputConnectionTrackingData() {
      if (getFBInterfaceSpec().mEITypeNames != nullptr) {
        mInputEventConnectionCount = std::make_unique<size_t[]>(getFBInterfaceSpec().mNumEIs);
      }
    }
```
or is there a reason why we would pass different values here?

Reply:
you are right there is no reason i missed that, thx